Atomic/atomic.py

```diff
@@ -326,12 +326,15 @@ def _pull_dockertar_layers(self, repo, imagebranch, temp_dir, input_layers):
     def _pull_image_to_ostree(self, repo, image, upgrade):
         if image.startswith("ostree:"):
             self._check_system_ostree_image(repo, image, upgrade)
-        elif self.args.image.startswith("docker:"):
-            self._pull_docker_image(repo, image.replace("docker:", ""))
+        elif self.args.image.startswith("oci:"):
+            self._check_system_oci_image(repo, image, upgrade)
         elif self.args.image.startswith("dockertar:"):
             self._pull_docker_tar(repo, image.replace("dockertar:", ""))
-        else: # Assume "oci:"
-            self._check_system_oci_image(repo, image, upgrade)
+        else: # Assume "docker:"
+            image = image.replace("docker:", "")
+            if ":" in image:
+                raise ValueError("Error: improper registry %s. Proper registries are docker, dockertar, oci, and ostree." % image)
+            self._pull_docker_image(repo, image)
 
```

Comment:
This chunk must be removed, the code was already doing the pull from the registry by default, this is the behaviour with "oci:".

"docker:" means to do the pull from the local Docker instance, that internally is docker save + import into OSTree.